test/react/hooks/use-room-reactions.test.tsx

```diff
@@ -0,0 +1,113 @@
+import { ConnectionLifecycle, Room, RoomLifecycle } from '@ably/chat';
+import { act, renderHook } from '@testing-library/react';
+import { beforeEach, describe, expect, it, vi } from 'vitest';
+
+import { useRoomReactions } from '../../../src/react/hooks/use-room-reactions.ts';
+import { makeRandomRoom } from '../../helper/room.ts';
+
+let mockRoom: Room;
+
+// apply mocks for the useChatConnection and useRoom hooks
+vi.mock('../../../src/react/hooks/use-chat-connection.js', () => ({
+  useChatConnection: () => ({ currentStatus: ConnectionLifecycle.Connected }),
+}));
+
+vi.mock('../../../src/react/hooks/use-room.js', () => ({
+  useRoom: () => ({ room: mockRoom, roomStatus: RoomLifecycle.Attached }),
+}));
+
+vi.mock('ably');
+
+describe('useRoomReactions', () => {
+  beforeEach(() => {
+    // create a new mock room before each test
+    vi.resetAllMocks();
+    mockRoom = makeRandomRoom({ options: { reactions: true } });
+  });
+
+  it('it should provide the room reactions instance and correct chat status response metrics', () => {
+    const { result } = renderHook(() => useRoomReactions());
+
+    // check that the room reactions instance is correctly provided
+    expect(result.current.reactions).toBe(mockRoom.reactions);
+
+    // check connection and room metrics are correctly provided
+    expect(result.current.roomStatus).toBe(RoomLifecycle.Attached);
+    expect(result.current.roomError).toBeUndefined();
+    expect(result.current.connectionStatus).toEqual(ConnectionLifecycle.Connected);
+    expect(result.current.connectionError).toBeUndefined();
+  });
+
+  it('it should correctly call the room reactions send method', async () => {
```

Comment:
It would be good to have at least one integration test that tests the entire flow from end-end